data/types/base/pump/basal_schedule.go

```diff
@@ -33,14 +33,27 @@ func ParseBasalSchedule(parser data.ObjectParser) *BasalSchedule {
 	return basalSchedule
 }
 
-func ParseBasalScheduleArray(parser data.ArrayParser) *map[string][]*BasalSchedule {
-	var basalScheduleArray *map[string][]*BasalSchedule
+func ParseBasalScheduleArray(parser data.ArrayParser) *[]*BasalSchedule {
+	var basalScheduleArray *[]*BasalSchedule
 	if parser.Array() != nil {
-		basalScheduleArray = &map[string][]*BasalSchedule{}
+		basalScheduleArray = &[]*BasalSchedule{}
 
-		// for index := range *parser.Array() {
-		// 	*basalScheduleArray[key] = ParseBasalSchedule(parser.NewChildObjectParser(key))
-		// }
+		for index := range *parser.Array() {
+			*basalScheduleArray = append(*basalScheduleArray, parseScheduleItem(parser.NewChildObjectParser(index)))
+		}
 	}
 	return basalScheduleArray
 }
+
+func ParseBasalSchedulesMap(parser data.ObjectParser) *map[string]*[]*BasalSchedule {
+
+	var basalScheduleMap map[string]*[]*BasalSchedule
+
+	if parser.Object() != nil {
```

Comment:
Since this function can return nil, do we validate that it isn't nil during the Validate phase?